Add document-question-answering task (#395)
huggingface/hub-docs: tasks/src/document-question-answering/about.md

## Use Cases

> Document question answering models can be used to answer natural language questions directly from document images. This is useful when the visual information of the document is important for answering the question.

Reviewer: not sure we want to mention this here -- but the models are also able to answer questions about documents _without_ images. E.g. the LayoutLM variants can take text + bounding boxes as input, so word documents, excel spreadsheets, pdfs with text, etc. can be processed w/ their inherent structure and w/out having to create an image.

Reviewer 2: I think we can have two use cases one including from PDFs another from text.

> Document question answering models can be used to answer natural language questions directly from document images. This is useful when the visual information of the document is important for answering the question.

Reviewer 3: Following the suggestion from @ankrgyl 
```suggestion
Document question-answering models can be used to answer natural language questions directly from a source document. Typically, document QA models can analyse a pictorial representation of a document (eg. a screenshot of an invoice). This is useful when the visual information of the document is important for answering the question. Nevertheless, certain document QA models, like the LayoutLM variants can work without images, taking the text and bounding boxes as input. This latter use case is useful for Word documents, Excel Spreadsheets, PDFs, etc.
```

### Document Parsing

You can use Document question answering models to parse forms. For example, you can extract the name, address, and other information from a form. You can also use the model to extract information from a table, or even a resume.

### Invoice Information Extraction

> You can use Document question answering models to extract information from invoices. For example, you can extract the invoice number, the invoice date, the total amount, the VAT number, and the invoice recipient.

Reviewer: Might want to provide some detail about why this is called out, e.g. "One of the most popular use cases is ... "

> You can use Document question answering models to parse forms. For example, you can extract the name, address, and other information from a form. You can also use the model to extract information from a table, or even a resume.
> 
> ### Invoice Information Extraction
> 
> You can use Document question answering models to extract information from invoices. For example, you can extract the invoice number, the invoice date, the total amount, the VAT number, and the invoice recipient.

Reviewer 3: ```suggestion
One of the most popular use cases of question-answering models is to parse forms and other kinds of structured documents. For example, you can extract the name, address, and other information from a form, a table, or even a resume.

### Invoice Information Extraction

Another very popular use case is the extraction of information from invoices. For example, you can extract the invoice number, the invoice date, the total amount, the VAT number, and the invoice recipient.
```

## Inference

> You can infer with Document QA models with the 🤗 Transformers library using the [`document-question-answering` pipeline](https://huggingface.co/docs/transformers/en/main_classes/pipelines#transformers.DocumentQuestionAnsweringPipeline). If no model checkpoint is given, the pipeline will be initialized with [`impira/layoutlm-document-qa`](https://huggingface.co/impira/layoutlm-document-qa). This pipeline answers the question(s) given as inputs by using the document(s). A document is defined as an image and an optional list of (word, box) tuples which represent the text in the document. If the word_boxes are not provided, it will use the Tesseract OCR engine (if available) to extract the words and boxes automatically for LayoutLM-like models which require them as input. For Donut, no OCR is run (see [paper](https://arxiv.org/abs/2111.15664)).

Reviewer 2: Can you avoid from using passive voice? e.g. maybe simplify This pipeline answers the question(s) given as inputs by using the document(s).

Author: yes! sorry about that, hope it's better now

## Use Cases

Document Question Answering models can be used to answer natural language questions about documents. Typically, document QA models consider textual, layout and potentially visual information. This is useful when the question requires some understanding of the visual aspects of the document.

> Nevertheless, certain document QA models can work without document images. Hence the task is not limited to visually-rich documents and allows to ask questions on spreadsheets, text PDFs, etc!

Reviewer 3: ```suggestion
Nevertheless, certain document QA models can work without document images. Hence the task is not limited to visually-rich documents and allows users to ask questions based on spreadsheets, text PDFs, etc!
```

### Document Parsing

One of the most popular use cases of document question answering models is the parsing of structured documents. For example, you can extract the name, address, and other information from a form. You can also use the model to extract information from a table, or even a resume.

### Invoice Information Extraction

Another very popular use case is invoice information extraction. For example, you can extract the invoice number, the invoice date, the total amount, the VAT number, and the invoice recipient.

## Inference

You can infer with Document QA models with the 🤗 Transformers library using the [`document-question-answering` pipeline](https://huggingface.co/docs/transformers/en/main_classes/pipelines#transformers.DocumentQuestionAnsweringPipeline). If no model checkpoint is given, the pipeline will be initialized with [`impira/layoutlm-document-qa`](https://huggingface.co/impira/layoutlm-document-qa). This pipeline takes question(s) and document(s) as input. 

> In this case, we define a document as an image and an optional list of (word, box) tuples representing the text in the document. The pipeline will then return the answer to the question in the document

Reviewer 3: Adding two trailing spaces so markdown keeps the text that follows on the next line.
```suggestion
In this case, we define a document as an image and an optional list of (word, box) tuples representing the text in the document. The pipeline will then return the answer to the question in the document.  
```

> In this case, we define a document as an image and an optional list of (word, box) tuples representing the text in the document. The pipeline will then return the answer to the question in the document

Reviewer 3: I don't quite understand what you mean by 

> "an optional list of (word, box) tuples"

Could you clarify?

Author: @KayO-GH this definition is taken from the Document QA pipeline [documentation](https://huggingface.co/docs/transformers/main_classes/pipelines#transformers.DocumentQuestionAnsweringPipeline).

Some document qa models (e.g. LayoutLM family) expect a list of words and associated bounding boxes as input. This is usually the output of the OCR (supported by the pipeline), but they can **optionally** be provided in the pipeline args (for example, when you already have the output of a third-party OCR).
I agree that this could be clearer, and might be a little bit cryptic to a person seeing these for the first time. Will try to simplify 😊
